iceoryx_posh/include/iceoryx_posh/capro/service_description.hpp

```diff
@@ -83,15 +84,17 @@ class ServiceDescription
   public:
     struct ClassHash
     {
-        ClassHash(const std::initializer_list<uint32_t>& values) noexcept;
-        uint32_t& operator[](const uint64_t index) noexcept;
-        const uint32_t& operator[](const uint64_t index) const noexcept;
-        bool operator==(const ClassHash& rhs) const noexcept;
-        bool operator!=(const ClassHash& rhs) const noexcept;
-
       private:
         static constexpr size_t CLASS_HASH_ELEMENT_COUNT{4u};
         uint32_t data[CLASS_HASH_ELEMENT_COUNT];
+
+      public:
```

Comment:
Why is it arranged in that way? In a struct everything is public until it is not stated as private. If you move this up then everything is public again.

Reply:
Yeah, Actually I have used "CLASS_HASH_ELEMENT_COUNT" macro in declaration of operator[], with previous arrangement it was no able to find that macro.

Reply:
The cleaner way is to put `CLASS_HASH_ELEMENT_COUNT` near to the other defined constexpr variables in namespace `iox::capro` at the top of the file. Then you don't need to reorder private/public defines and it can be used by the end-user for cleanly iterating over the ClassHash.